examples/routing_service/routing_service_msql_adapter/cpp/ConnectorMsqlDb.cxx

```diff
@@ -0,0 +1,337 @@
+ * warranty of any type, including any warranty for fitness for any purpose.
+ * RTI is under no obligation to maintain or support the Software.  RTI shall
+ * not be liable for any incidental or consequential damages arising out of the
+ * use or inability to use the software.
+ */
+
+#include "ConnectorMsqlDb.hpp"
+#include "RsLog.hpp"
+#include <sqlext.h>
+
+#define RTI_RS_LOG_ARGS "ConnectorMsqlDb"
+#define SQL_BUFF_LEN 1024
+#define TIMEOUT 5
+
+using namespace rti::routing;
+using namespace dds::core::xtypes;
+
+ConnectorMsqlDb::ConnectorMsqlDb(
+        const std::string topicName,
+        const PropertySet &properties)
+        : ConnectorBase(topicName)
+{
+    setConnectionInfo(properties);
+    connect();
+}
+
+ConnectorMsqlDb::~ConnectorMsqlDb()
+{
+    disconnect();
+}
+
+/*
+ * Connect to the MSQL database.
+ * Return true if connection succeeds, otherwise false
+ */
+auto ConnectorMsqlDb::connect() -> bool
+{
+    RTI_RS_LOG_FN(connect_db);
+
+    SQLRETURN retcode;
+
+    /* Allocates the environment handle */
+    retcode = SQLAllocHandle(SQL_HANDLE_ENV, SQL_NULL_HANDLE, &sqlEnvHandle);
+    if (retcode != SQL_SUCCESS && retcode != SQL_SUCCESS_WITH_INFO) {
+        RTI_RS_ERROR("SQL Error - Allocating the environment handle failed");
+        disconnect();
+        return connected_;
+    }
+
+    /* Set the ODBC version environment attribute */
+    retcode = SQLSetEnvAttr(
+            sqlEnvHandle,
+            SQL_ATTR_ODBC_VERSION,
+            (SQLPOINTER) SQL_OV_ODBC3,
+            0);
+    if (retcode != SQL_SUCCESS && retcode != SQL_SUCCESS_WITH_INFO) {
+        RTI_RS_ERROR(
+                "SQL Error - Setting the ODBC version environment attribute");
+        disconnect();
+        return connected_;
+    }
+
+    /* Allocates the connection handle */
+    retcode = SQLAllocHandle(SQL_HANDLE_DBC, sqlEnvHandle, &sqlConnHandle);
+    if (retcode != SQL_SUCCESS && retcode != SQL_SUCCESS_WITH_INFO) {
+        RTI_RS_ERROR("SQL Error - Allocating SQL database connection failed");
+        disconnect();
+        return connected_;
+    }
+
+    /* Sets login timeout to 5 seconds */
+    retcode = SQLSetConnectAttr(
+            sqlConnHandle,
+            SQL_LOGIN_TIMEOUT,
+            (SQLPOINTER) TIMEOUT,
+            0);
+    if (retcode != SQL_SUCCESS && retcode != SQL_SUCCESS_WITH_INFO) {
+        RTI_RS_ERROR("SQL Error - Setting connection timeout failed");
+        disconnect();
+        return connected_;
+    }
+
+    SQLCHAR retconstring[SQL_BUFF_LEN];
+    /* connect to SQL Server */
+    retcode = SQLDriverConnect(
+            sqlConnHandle,
+            NULL,
+            (SQLCHAR *) connectionInfo.data(),
+            SQL_NTS,
+            retconstring,
+            SQL_BUFF_LEN,
+            NULL,
+            SQL_DRIVER_NOPROMPT);
+
+    if (retcode == SQL_ERROR) {
+        msqlShowError(SQL_HANDLE_DBC, sqlConnHandle);
+        disconnect();
+        return connected_;
+    } else {
```

Comment:
no need for else (you're returning within if). Review the code for other places like this

Reply:
Done